HW1/P6/P6.py

```diff
@@ -0,0 +1,109 @@
+import random
+
+import findspark
+findspark.init('/home/lhoang/spark')
+
+import pyspark
+sc = pyspark.SparkContext(appName="spark1")
+
+
+def get_gram(v):
+    '''
+    Converts an iterable into a sorted dictionary mapping from value to
+    number of occurences.
+    :param v: the iterable to convert
+    '''
+    D = {}
+    for vv in v:
+        if vv in D:
+            D[vv] += 1
+        else:
+            D[vv] = 1
+    a = []
+    for k in D:
+        a.append((k, D[k]))
+    return sorted(a, key=lambda x: -x[1])
+
+
+def is_number(s):
+    '''
+    Checks if the given string is a number.
+    :param s: the string to check
+    '''
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+
+def generate_next_word(kv):
+    '''
+    Generate next word by sampling from the list of possible
+    next words based on their counts.
+    '''
+    sum_occurences = 0
+    for v in kv[1][0]:
+        sum_occurences += v[1]
+
+    random_index = random.randint(1, sum_occurences)
+
+    next_word = kv[1][0][0][0]
```

Comment:
hmmm...all of this indexing shouldn't be necessary. maybe remove a layer of listing from your generation of kv pairs?